### PR TITLE
Upgrade socket io

### PR DESCRIPTION
**Pull request by Eirika**

### CODE REVIEW

File: player/models.py [hunk 1]
```diff
@@ -130,6 +130,11 @@ def push(self, music_id, requestId=None, **kwargs):
             self.current_music = music
             self.save()
             self.play_next(forced=True)
+        else:
```
Comment (Amoki) [inline player/models.py]: This piece of code is really confuse and not documented...
the push function just push music in the queue (and start the music if the queue is empty)
I should refactor this (it's easy)

Your code should be in the play function

Comment (Amoki) [inline player/models.py]: factorize this in a update function.
example: self.update()

File: player/models.py [hunk 2]
```diff
@@ -196,9 +201,18 @@ def toggle_shuffle(self, to_active):
             self.save()
             if not self.current_music:
                 self.play_next()
+            else:
```
Comment (Amoki) [inline player/models.py]: same

File: player/models.py [hunk 3]
```diff
@@ -196,9 +201,18 @@ def toggle_shuffle(self, to_active):
...
+                message = {
+                    'action': 'update',
+                }
+                self.send_message(message)
         else:
             self.shuffle = False
             self.save()
+            message = {
```
Comment (Amoki) [inline player/models.py]: same

Comment (Amoki) [inline player/models.py]: I think we can do the same action in a full pythonic way....
Je t'en parle demain moins bourré

File: player/views/client/client_player.py [hunk 4]
```diff
@@ -47,8 +48,30 @@ def next_music(request):
...
+
+        musics = room.music_set.all().order_by('-date')
+        # Get the paginator
+        paginator = Paginator(musics, 16)
+        more_musics = False
+        try:
+            page = int(request.POST.get('page')) - 1
+        except ValueError:
+            page = 1
+            print("Error on page requested")
```
Comment (Amoki) [inline player/views/client/client_player.py]: send an error to the client or skip the request. We don't care about log bastards that send us shit :p

File: player/views/client/client_player.py [hunk 5]
```diff
@@ -47,8 +48,30 @@ def next_music(request):
...
+        try:
+            musics = []
+            for i in range(0, page):
+                musics += paginator.page(i + 1).object_list
```
Comment (Amoki) [inline player/views/client/client_player.py]: there is no way to check in page i + 1 exists instead of trying it ?

File: player/models.py (hunk 1 above)
Comment (Eirika) [inline player/models.py]: Fixed

File: player/models.py [hunk 6]
```diff
@@ -130,6 +130,8 @@ def push(self, music_id, requestId=None, **kwargs):
             self.current_music = music
             self.save()
             self.play_next(forced=True)
+        else:
+            send_update_message()
```
Comment (Amoki) [inline player/models.py]: where is this function ? :p

Comment (Amoki) [inline player/models.py]: self.

File: static/js/remote/ajax.js [hunk 7]
```diff
@@ -239,9 +216,19 @@ function update_player() {
...
 }
+
+function log_errors(resultat, statut, erreur){
+  console.log(resultat.responseText);
+  console.log("Statut : "+statut);
+  console.log("Error : "+erreur);
+}
+
+socket.on('message', function(message) {
+  if(message.action !== 'volume_up' && message.action !== 'volume_down'){
```
Comment (Amoki) [inline static/js/remote/ajax.js]: if message.action === 'update' ?

**Pull request merged**